syntax = "proto2";
package hw.trezor.messages.filecoin;

// Sugar for easier handling in Java
option java_package = "com.satoshilabs.trezor.lib.protobuf";
option java_outer_classname = "TrezorMessageFilecoin";

/**
 * Request: Address at the specified index
 * @start
 * @next FilecoinAddress
 */
message FilecoinGetAddress {
    repeated uint32 address_n = 1;              // BIP-32 path to derive the key from master node
    optional bool show_display = 2;             // optionally show on display before sending the result
    optional bool testnet = 3;                  // default main network
}

/**
 * Response: Address for the given index
 * @end
 */
message FilecoinAddress {
    optional string address = 1;                // Filecoin address as hex-encoded string
}

/**
 * Request: ask device to sign Filecoin transaction
 * @start
 * @next FilecoinSignedTx
 */
message FilecoinSignTx {
    repeated uint32 address_n = 1;         // BIP-32 path to derive the key from master node
    required bytes raw_tx = 2;             // serialized raw transaction
    optional bool testnet = 3;             // default main network
}

/**
 * Response: signature for transaction
 * @end
 */
message FilecoinSignedTx {
    required bytes signature = 1;           // the signature of the raw transaction
}
